code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -447,18 +435,48 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 		new /obj/item/stack/sheet/iron (drop_location(), 5)
 		qdel(src)
 		return
-	switch(mode)
-		if(CANISTER_TIER_1)
-			new /obj/item/stack/sheet/iron (drop_location(), 10)
-		if(CANISTER_TIER_2)
-			new /obj/item/stack/sheet/iron (drop_location(), 10)
-			new /obj/item/stack/sheet/plasteel (drop_location(), 5)
-		if(CANISTER_TIER_3)
-			new /obj/item/stack/sheet/iron (drop_location(), 10)
-			new /obj/item/stack/sheet/plasteel (drop_location(), 5)
-			new /obj/item/stack/sheet/bluespace_crystal (drop_location(), 1)
+	new /obj/item/stack/sheet/iron (drop_location(), 10)
+	if(internal_cell)
+		internal_cell.forceMove(drop_location())
 	qdel(src)
 
+/obj/machinery/portable_atmospherics/canister/attackby(obj/item/item, mob/user, params)
+	if(istype(item, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/active_cell = item
+		if(!cell_container_opened)
+			balloon_alert(user, "open the hatch first")
+			return
+		if(internal_cell)
+			if(!user.transferItemToLoc(active_cell, src))
+				return
+			user.put_in_hands(internal_cell)
+			internal_cell = active_cell
+			balloon_alert(user, "you successfully replace the cell")
+			return
+		if(!user.transferItemToLoc(active_cell, src))
+			return
+		internal_cell = active_cell
+		balloon_alert(user, "you successfully install the cell")
+		return
+	return ..()
+
+/obj/machinery/portable_atmospherics/canister/screwdriver_act(mob/living/user, obj/item/screwdriver)
+	if(screwdriver.tool_behaviour != TOOL_SCREWDRIVER)
+		return
+	screwdriver.play_tool_sound(src, 50)
+	cell_container_opened = !cell_container_opened
+	to_chat(user, span_notice("You [cell_container_opened ? "open" : "close"] the cell container hatch of [src]."))
+	update_appearance()
+	return TRUE
```

Comment:
didn't we decon with a screwdriver before? or am I deranged

Reply:
we used the welder to decon it, the screwdriver was on the frame to complete the canister